packages/lesmis/src/routes/AdminPanelApp.js

```diff
@@ -143,46 +145,76 @@ export const ROUTES = [
   },
 ];
 
-const config = { apiUrl: `${getApiUrl()}/admin` };
-
-const HeaderContainer = styled.div`
-  background: ${props => props.theme.palette.primary.main};
-  border-top: 1px solid rgba(0, 0, 0, 0.2); ;
-`;
-
-const AdminPanelRoutes = () => {
+const AdminPanelApp = ({ user, isBESAdmin }) => {
   const headerEl = React.useRef(null);
   const { path } = useRouteMatch();
-  const { isLesmisAdmin } = useUser();
+  const adminUrl = useAdminPanelUrl();
 
   const getHeaderEl = () => {
     return headerEl;
   };
 
+  const routes = getRoutes(adminUrl);
+
   return (
-    <AdminPanelDataProviders config={config}>
-      <div>
-        <HeaderContainer ref={headerEl} />
+    <Switch>
+      <Route path={`${path}/login`} exact>
+        <AdminPanelLoginPage />
+      </Route>
+      <Route path={`${path}/logout`} exact>
+        <LogoutPage redirectTo={`${adminUrl}/login`} />
+      </Route>
+      <LesmisAdminRoute key={`${adminUrl}/viz-builder`} path={`${path}/viz-builder`} isBESAdmin>
+        <VizBuilderProviders>
+          <VizBuilderApp
+            basePath={adminUrl}
+            Navbar={({ user: vizBuilderUser }) => <AdminPanelNavbar user={vizBuilderUser} />}
+          />
+        </VizBuilderProviders>
+      </LesmisAdminRoute>
+      <PrivateRoute path={`${path}`} loginPath={`${adminUrl}/login`}>
+        <AdminPanelNavbar user={user} links={routes} />
+        <div ref={headerEl} />
         <Switch>
-          {[...ROUTES].map(route => (
-            <LesmisAdminRoute key={route.to} path={`${path}${route.to}`}>
+          {[...routes].map(route => (
+            <LesmisAdminRoute key={route.to} path={`${route.to}`} isBESAdmin={isBESAdmin}>
```

Comment:
Once @chris-pollard's changes come in we won't need this `isBESAdmin` stuff, so not really worried about it here 👍